syntax = "proto3";

package panda_studio.panda_agent;
option go_package = "github.com/panda-re/panda_studio/panda_agent/pb";

service PandaExecutor {
    rpc BootMachine (BootMachineRequest) returns (BootMachineReply) {}
    rpc Shutdown (ShutdownRequest) returns (ShutdownReply) {}
    rpc RunCommand (RunCommandRequest) returns (RunCommandReply) {}
}

message BootMachineRequest {}
message BootMachineReply {}
message ShutdownRequest {}
message ShutdownReply {}

message RunCommandRequest {
    string command = 1;
}

message RunCommandReply {
    int32 statusCode = 1;
    string output = 2;
}
